File: app/src/main/java/com/explead/seasons/winter/logic/WinterArrow.java
package com.explead.seasons.winter.logic;

import com.explead.seasons.common.logic.Direction;

public class WinterArrow extends Cell {

    private int id;
    private Direction direction;

    public WinterArrow(int x, int y, int id) {
        super(x, y);
        this.id = id;
        findDirection();
    }

    public void findDirection() {
        switch (id) {
            case 1:
                direction = Direction.U;
                break;
            case 2:
                direction = Direction.R;
                break;
            case 3:
                direction = Direction.D;
                break;
            case 4:
                direction = Direction.L;
                break;
        }
    }

    public Direction getDirection() {
        return direction;
    }

    public int getId() {
        return id;
    }

    public char gtNameId() {
        return direction.name().charAt(0);
    }
}

File: app/src/main/java/com/explead/seasons/winter/logic/WinterArrow.kt
package com.explead.seasons.winter.logic

import com.explead.seasons.common.logic.Cell
import com.explead.seasons.common.logic.Direction

class WinterArrow(x: Int, y: Int, val id: Int) : Cell(x, y) {
    var direction: Direction? = null
        private set

    init {
        findDirection()
    }

    private fun findDirection() {
        when (id) {
            1 -> direction = Direction.U
            2 -> direction = Direction.R
            3 -> direction = Direction.D
            4 -> direction = Direction.L
        }
    }

    fun gtNameId(): Char {
        return direction?.name?.get(0)!!
    }
}
